### ISSUE TITLE
bear fails to build

### ISSUE THREAD

**Issue by dbeecham** [NixOS/nixpkgs#103929]
**Describe the bug**

`bear` fails to build with the following log

```
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/rqx4ijv0y3mp2rh8fx1gr7332k2iydh1-source
source root is source
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
applying patch /nix/store/mb32kqai6asfrv9is962ays20685xqn3-no-double-relative.patch
patching file source/config.h.in
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
fixing cmake files...
cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DCMAKE_SKIP_BUILD_RPATH=ON -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/5gsbr4a1vhzbd3jkkc3d3dksn096rs10-bear-3.0.2/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/5gsbr4a1vhzbd3jkkc3d3dksn096rs10-bear-3.0.2/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/5gsbr4a1vhzbd3jkkc3d3dksn096rs10-bear-3.0.2/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/5gsbr4a1vhzbd3jkkc3d3dksn096rs10-bear-3.0.2/share/doc/Bear -DCMAKE_INSTALL_INFODIR=/nix/store/5gsbr4a1vhzbd3jkkc3d3dksn096rs10-bear-3.0.2/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/5gsbr4a1vhzbd3jkkc3d3dksn096rs10-bear-3.0.2/share/man -DCMAKE_INSTALL_OLDINCLUDEDIR=/nix/store/5gsbr4a1vhzbd3jkkc3d3dksn096rs10-bear-3.0.2/include -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/5gsbr4a1vhzbd3jkkc3d3dksn096rs10-bear-3.0.2/include -DCMAKE_INSTALL_SBINDIR=/nix/store/5gsbr4a1vhzbd3jkkc3d3dksn096rs10-bear-3.0.2/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/5gsbr4a1vhzbd3jkkc3d3dksn096rs10-bear-3.0.2/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/5gsbr4a1vhzbd3jkkc3d3dksn096rs10-bear-3.0.2/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_OSX_SYSROOT= -DCMAKE_OSX_ARCHITECTURES=x86_64 -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/store/s59zibwwm8vy5ymbg5c3d0aksl8lz4h2-binutils-2.31.1/bin/strip -DCMAKE_RANLIB=/nix/store/s59zibwwm8vy5ymbg5c3d0aksl8lz4h2-binutils-2.31.1/bin/ranlib -DCMAKE_AR=/nix/store/s59zibwwm8vy5ymbg5c3d0aksl8lz4h2-binutils-2.31.1/bin/ar -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_INSTALL_PREFIX=/nix/store/5gsbr4a1vhzbd3jkkc3d3dksn096rs10-bear-3.0.2  
-- The C compiler identification is GNU 9.3.0
-- The CXX compiler identification is GNU 9.3.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /nix/store/lfl1davviy5mai3g80mpy2gkb97rh9j5-gcc-wrapper-9.3.0/bin/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/lfl1davviy5mai3g80mpy2gkb97rh9j5-gcc-wrapper-9.3.0/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for nlohman_json dependency
-- Looking for nlohman_json dependency -- found
-- Looking for fmt dependency
-- Looking for fmt dependency -- found
-- Looking for spdlog dependency
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Looking for spdlog dependency -- found
-- Looking for gRPC::grpc++ dependency
-- Found PkgConfig: pkg-config (found version "0.29.2") 
-- Checking for modules 'protobuf>=3.11;grpc++>=1.28'
--   Found protobuf, version 3.13.0
--   Found grpc++, version 1.33.2
-- Looking for gRPC::grpc++ dependency -- found
-- Looking for protoc
-- Looking for protoc -- found
-- Looking for grpc_cpp_plugin
-- Looking for grpc_cpp_plugin -- found
-- Looking for GTest dependency
-- Checking for modules 'gtest>=1.10;gtest_main>=1.10;gmock>=1.10'
--   Found gtest, version 1.10.0
--   Found gtest_main, version 1.10.0
--   Found gmock, version 1.10.0
-- Looking for GTest dependency -- found
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    BUILD_TESTING
    CMAKE_EXPORT_NO_PACKAGE_REGISTRY
    CMAKE_POLICY_DEFAULT_CMP0025


-- Build files have been written to: /build/source/build
cmake: enabled parallel building
@nix { "action": "setPhase", "phase": "buildPhase" }
building
build flags: -j4 -l4 SHELL=/nix/store/pmrhk324fkidrm5ffd5jckb21s9zys6r-bash-4.4-p23/bin/bash
[35m[1mScanning dependencies of target nlohmann_json_dependency[0m
[35m[1mScanning dependencies of target spdlog_dependency[0m
[35m[1mScanning dependencies of target fmt_dependency[0m
[35m[1mScanning dependencies of target googletest_dependency[0m
[  0%] Built target fmt_dependency
[  0%] Built target spdlog_dependency
[  0%] Built target nlohmann_json_dependency
[  0%] Built target googletest_dependency
[35m[1mScanning dependencies of target grpc_dependency[0m
[  0%] Built target grpc_dependency
[35m[1mScanning dependencies of target BearSource[0m
[  5%] [34m[1mCreating directories for 'BearSource'[0m
[ 11%] [34m[1mNo download step for 'BearSource'[0m
[ 16%] [34m[1mNo update step for 'BearSource'[0m
[ 22%] [34m[1mNo patch step for 'BearSource'[0m
[ 27%] [34m[1mPerforming configure step for 'BearSource'[0m
loading initial cache file /build/source/build/subprojects/tmp/BearSource/BearSource-cache-Release.cmake
-- The C compiler identification is GNU 9.3.0
-- The CXX compiler identification is GNU 9.3.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /nix/store/lfl1davviy5mai3g80mpy2gkb97rh9j5-gcc-wrapper-9.3.0/bin/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/lfl1davviy5mai3g80mpy2gkb97rh9j5-gcc-wrapper-9.3.0/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found PkgConfig: pkg-config (found version "0.29.2") 
-- Checking for modules 'gtest;gtest_main;gmock'
--   Found gtest, version 1.10.0
--   Found gtest_main, version 1.10.0
--   Found gmock, version 1.10.0
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Found nlohmann_json: /nix/store/3qcgwdk875n4qqmzi23z5xdb752r5wll-nlohmann_json-3.7.3/lib/cmake/nlohmann_json/nlohmann_jsonConfig.cmake (found version "3.7.3") 
-- Checking for modules 'protobuf;grpc++'
--   Found protobuf, version 3.13.0
--   Found grpc++, version 1.33.2
-- Looking for spawn.h
-- Looking for spawn.h - found
-- Looking for unistd.h
-- Looking for unistd.h - found
-- Looking for dlfcn.h
-- Looking for dlfcn.h - found
-- Looking for errno.h
-- Looking for errno.h - found
-- Looking for sys/utsname.h
-- Looking for sys/utsname.h - found
-- Looking for sys/wait.h
-- Looking for sys/wait.h - found
-- Looking for gnu/lib-names.h
-- Looking for gnu/lib-names.h - found
-- Looking for _NSGetEnviron
-- Looking for _NSGetEnviron - not found
-- Looking for dlopen
-- Looking for dlopen - found
-- Looking for dlsym
-- Looking for dlsym - found
-- Looking for dlerror
-- Looking for dlerror - found
-- Looking for dlclose
-- Looking for dlclose - found
-- Looking for RTLD_NEXT
-- Looking for RTLD_NEXT - found
-- Looking for EACCES
-- Looking for EACCES - found
-- Looking for ENOENT
-- Looking for ENOENT - found
-- Looking for strerror_r
-- Looking for strerror_r - found
-- Looking for environ
-- Looking for environ - found
-- Looking for uname
-- Looking for uname - found
-- Looking for confstr
-- Looking for confstr - found
-- Looking for _CS_PATH
-- Looking for _CS_PATH - found
-- Looking for _CS_GNU_LIBC_VERSION
-- Looking for _CS_GNU_LIBC_VERSION - found
-- Looking for _CS_GNU_LIBPTHREAD_VERSION
-- Looking for _CS_GNU_LIBPTHREAD_VERSION - found
-- Looking for protoc ... /nix/store/snbchia39jaw83fpikmq75kd5g11yrv0-protobuf-3.13.0/bin/protoc
-- Looking for grpc_cpp_plugin ... /nix/store/acqgjldh2msifsgl2xx0ifsay777i9rf-grpc-1.33.2/bin/grpc_cpp_plugin
-- Configuring done
-- Generating done
-- Build files have been written to: /build/source/build/subprojects/Build/BearSource
[ 33%] [34m[1mPerforming build step for 'BearSource'[0m
[35m[1mScanning dependencies of target sys_a[0m
[35m[1mScanning dependencies of target flags_a[0m
[35m[1mScanning dependencies of target report_a[0m
[35m[1mScanning dependencies of target shell_a[0m
[  1%] [32mBuilding CXX object libflags/CMakeFiles/flags_a.dir/source/Flags.cc.o[0m
[  2%] [32mBuilding CXX object libreport/CMakeFiles/report_a.dir/source/Report.cc.o[0m
[  3%] [32mBuilding CXX object libsys/CMakeFiles/sys_a.dir/source/Os.cc.o[0m
[  5%] [32mBuilding CXX object libshell/CMakeFiles/shell_a.dir/source/Command.cc.o[0m
[  6%] [32mBuilding CXX object libsys/CMakeFiles/sys_a.dir/source/Guard.cc.o[0m
[  6%] Built target flags_a
[35m[1mScanning dependencies of target result_unit_test[0m
[  7%] [32mBuilding CXX object libresult/CMakeFiles/result_unit_test.dir/test/ResultTest.cc.o[0m
[  8%] [32mBuilding CXX object libsys/CMakeFiles/sys_a.dir/source/Errors.cc.o[0m
[ 10%] [32mBuilding CXX object libsys/CMakeFiles/sys_a.dir/source/Path.cc.o[0m
[ 11%] [32mBuilding CXX object libsys/CMakeFiles/sys_a.dir/source/Process.cc.o[0m
[ 12%] [32m[1mLinking CXX executable result_unit_test[0m
[ 12%] Built target result_unit_test
[ 13%] [32mBuilding CXX object libsys/CMakeFiles/sys_a.dir/source/Signal.cc.o[0m
[ 13%] Built target shell_a
[35m[1mScanning dependencies of target flags_unit_test[0m
[ 15%] [32mBuilding CXX object libflags/CMakeFiles/flags_unit_test.dir/test/FlagsTest.cc.o[0m
[35m[1mScanning dependencies of target shell_unit_test[0m
[ 16%] [32mBuilding CXX object libshell/CMakeFiles/shell_unit_test.dir/test/CommandTest.cc.o[0m
[ 16%] Built target report_a
[ 17%] [34m[1mGenerating supervise.pb.h, supervise.grpc.pb.h, supervise.pb.cc, supervise.grpc.pb.cc[0m
[35m[1mScanning dependencies of target supervise_a[0m
[ 18%] [32mBuilding CXX object intercept/proto/CMakeFiles/supervise_a.dir/supervise.pb.cc.o[0m
[ 20%] [32m[1mLinking CXX executable flags_unit_test[0m
[ 20%] Built target flags_unit_test
[35m[1mScanning dependencies of target exec_a[0m
[ 21%] [32mBuilding CXX object intercept/CMakeFiles/exec_a.dir/source/report/libexec/Buffer.cc.o[0m
[ 22%] [32m[1mLinking CXX executable shell_unit_test[0m
[ 24%] [32mBuilding CXX object intercept/CMakeFiles/exec_a.dir/source/report/libexec/Environment.cc.o[0m
[ 24%] Built target shell_unit_test
[35m[1mScanning dependencies of target report_unit_test[0m
[ 25%] [32mBuilding CXX object libreport/CMakeFiles/report_unit_test.dir/test/ReportTest.cc.o[0m
[ 26%] [32mBuilding CXX object intercept/CMakeFiles/exec_a.dir/source/report/libexec/Executor.cc.o[0m
[ 27%] [32mBuilding CXX object intercept/CMakeFiles/exec_a.dir/source/report/libexec/Linker.cc.o[0m
[ 29%] [32mBuilding CXX object intercept/CMakeFiles/exec_a.dir/source/report/libexec/Logger.cc.o[0m
[ 29%] Built target sys_a
[35m[1mScanning dependencies of target bear[0m
[ 30%] [32mBuilding CXX object bear/CMakeFiles/bear.dir/main.cc.o[0m
[ 31%] [32mBuilding CXX object intercept/CMakeFiles/exec_a.dir/source/report/libexec/Paths.cc.o[0m
[ 32%] [32mBuilding CXX object intercept/CMakeFiles/exec_a.dir/source/report/libexec/Resolver.cc.o[0m
[ 34%] [32m[1mLinking CXX executable report_unit_test[0m
[ 34%] Built target report_unit_test
[ 35%] [32mBuilding CXX object intercept/CMakeFiles/exec_a.dir/source/report/libexec/Session.cc.o[0m
[35m[1mScanning dependencies of target citnames_json_a[0m
[ 36%] [32mBuilding CXX object citnames/CMakeFiles/citnames_json_a.dir/source/Configuration.cc.o[0m
[ 36%] Built target exec_a
[35m[1mScanning dependencies of target citnames_a[0m
[ 37%] [32mBuilding CXX object citnames/CMakeFiles/citnames_a.dir/source/Application.cc.o[0m
[ 39%] [32mBuilding CXX object intercept/proto/CMakeFiles/supervise_a.dir/supervise.grpc.pb.cc.o[0m
[ 40%] [32m[1mLinking CXX executable bear[0m
/nix/store/s59zibwwm8vy5ymbg5c3d0aksl8lz4h2-binutils-2.31.1/bin/ld: CMakeFiles/bear.dir/main.cc.o: in function `std::_Function_handler<void (int const&), main::{lambda(auto:1)#4}>::_M_invoke(std::_Any_data const&, int const&)':
main.cc:(.text+0x4fd): undefined reference to `spdlog::details::log_msg::log_msg(spdlog::source_loc, fmt::v7::basic_string_view<char>, spdlog::level::level_enum, fmt::v7::basic_string_view<char>)'
/nix/store/s59zibwwm8vy5ymbg5c3d0aksl8lz4h2-binutils-2.31.1/bin/ld: CMakeFiles/bear.dir/main.cc.o: in function `auto main::{lambda(auto:1 const&)#1}::operator()<flags::Arguments>(flags::Arguments const&) const [clone .isra.0]':
main.cc:(.text+0xb16): undefined reference to `spdlog::details::log_msg::log_msg(spdlog::source_loc, fmt::v7::basic_string_view<char>, spdlog::level::level_enum, fmt::v7::basic_string_view<char>)'
/nix/store/s59zibwwm8vy5ymbg5c3d0aksl8lz4h2-binutils-2.31.1/bin/ld: main.cc:(.text+0xc0b): undefined reference to `spdlog::details::log_msg::log_msg(spdlog::source_loc, fmt::v7::basic_string_view<char>, spdlog::level::level_enum, fmt::v7::basic_string_view<char>)'
/nix/store/s59zibwwm8vy5ymbg5c3d0aksl8lz4h2-binutils-2.31.1/bin/ld: main.cc:(.text+0xd08): undefined reference to `spdlog::details::log_msg::log_msg(spdlog::source_loc, fmt::v7::basic_string_view<char>, spdlog::level::level_enum, fmt::v7::basic_string_view<char>)'
/nix/store/s59zibwwm8vy5ymbg5c3d0aksl8lz4h2-binutils-2.31.1/bin/ld: main.cc:(.text+0xdfd): undefined reference to `spdlog::details::log_msg::log_msg(spdlog::source_loc, fmt::v7::basic_string_view<char>, spdlog::level::level_enum, fmt::v7::basic_string_view<char>)'
/nix/store/s59zibwwm8vy5ymbg5c3d0aksl8lz4h2-binutils-2.31.1/bin/ld: CMakeFiles/bear.dir/main.cc.o:main.cc:(.text+0x471a): more undefined references to `spdlog::details::log_msg::log_msg(spdlog::source_loc, fmt::v7::basic_string_view<char>, spdlog::level::level_enum, fmt::v7::basic_string_view<char>)' follow
collect2: error: ld returned 1 exit status
make[5]: *** [bear/CMakeFiles/bear.dir/build.make:119: bear/bear] Error 1
make[4]: *** [CMakeFiles/Makefile2:1095: bear/CMakeFiles/bear.dir/all] Error 2
make[4]: *** Waiting for unfinished jobs....
[ 41%] [32mBuilding CXX object citnames/CMakeFiles/citnames_a.dir/source/semantic/Parsers.cc.o[0m
[ 43%] [32mBuilding CXX object citnames/CMakeFiles/citnames_a.dir/source/semantic/Semantic.cc.o[0m
[ 44%] [32mBuilding CXX object citnames/CMakeFiles/citnames_a.dir/source/semantic/Tool.cc.o[0m
[ 44%] Built target supervise_a
[ 45%] [32mBuilding CXX object citnames/CMakeFiles/citnames_a.dir/source/semantic/ToolClang.cc.o[0m
[ 46%] [32mBuilding CXX object citnames/CMakeFiles/citnames_json_a.dir/source/Output.cc.o[0m
[ 48%] [32mBuilding CXX object citnames/CMakeFiles/citnames_a.dir/source/semantic/ToolCuda.cc.o[0m
[ 49%] [32mBuilding CXX object citnames/CMakeFiles/citnames_a.dir/source/semantic/ToolGcc.cc.o[0m
[ 50%] [32mBuilding CXX object citnames/CMakeFiles/citnames_a.dir/source/semantic/ToolWrapper.cc.o[0m
[ 51%] [32mBuilding CXX object citnames/CMakeFiles/citnames_a.dir/source/semantic/ToolExtendingWrapper.cc.o[0m
[ 51%] Built target citnames_json_a
[ 51%] Built target citnames_a
make[3]: *** [Makefile:160: all] Error 2
make[2]: *** [CMakeFiles/BearSource.dir/build.make:133: subprojects/Stamp/BearSource/BearSource-build] Error 2
make[1]: *** [CMakeFiles/Makefile2:250: CMakeFiles/BearSource.dir/all] Error 2
make: *** [Makefile:171: all] Error 2
```

**To Reproduce**
1. `inputs.nixpkgs = { type = "github"; owner = "NixOS"; repo = "nixpkgs"; rev = "f3a787f948516dc8f8aa269c90cfe03ceb358465";};` - the latest commit on master at the time of writing this issue, or `nix registry add github:NixOS/nixpkgs nixpkgs`
2. `nix build` or `nix build nixpkgs#bear`
3. fail

**Expected behavior**
expect bear to build

**Screenshots**
No interesting screenshots

**Additional context**
```
> nix registry list
global flake:blender-bin github:edolstra/nix-warez
global flake:dwarffs github:edolstra/dwarffs
global flake:hydra github:NixOS/hydra
global flake:nimble github:nix-community/flake-nimble
global flake:nix github:NixOS/nix
global flake:nixops github:NixOS/nixops
global flake:nixos-hardware github:NixOS/nixos-hardware
global flake:nixos-homepage github:NixOS/nixos-homepage/flake
global flake:nur github:nix-community/NUR
global flake:nixpkgs/release-19.09 github:edolstra/nixpkgs/release-19.09
global flake:nixpkgs github:NixOS/nixpkgs
global flake:templates github:NixOS/templates
global flake:patchelf github:NixOS/patchelf

> nix build nixpkgs#bear -o bear
error: --- Error ------------------------------------------------------------------------------------------------ nix
error: --- Error --- nix-daemon
builder for '/nix/store/jm3jqgx9g6rac881sv52v2y40r9v2jr5-bear-3.0.2.drv' failed with exit code 2; last 10 log lines:
  [ 48%] Building CXX object citnames/CMakeFiles/citnames_a.dir/source/semantic/ToolCuda.cc.o
  [ 49%] Building CXX object citnames/CMakeFiles/citnames_a.dir/source/semantic/ToolGcc.cc.o
  [ 50%] Building CXX object citnames/CMakeFiles/citnames_a.dir/source/semantic/ToolWrapper.cc.o
  [ 51%] Building CXX object citnames/CMakeFiles/citnames_a.dir/source/semantic/ToolExtendingWrapper.cc.o
  [ 51%] Built target citnames_json_a
  [ 51%] Built target citnames_a
  make[3]: *** [Makefile:160: all] Error 2
  make[2]: *** [CMakeFiles/BearSource.dir/build.make:133: subprojects/Stamp/BearSource/BearSource-build] Error 2
  make[1]: *** [CMakeFiles/Makefile2:250: CMakeFiles/BearSource.dir/all] Error 2
  make: *** [Makefile:171: all] Error 2

```

**Notify maintainers**
@babariviere 

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.
Is this relevant for me if I use nix flakes? Anyway here is the output

```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.72, NixOS, 21.03.20201021.d2b8b92 (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 3.0pre20200829_f156513`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

**Comment by MetaDark** [comment]
It looks like this was fixed somewhere between commit 87e9b49fc785a6ff20080a1c80b35dfb95c8c222 and 7bea3c7976aa6cb4f9c3614bda2dbbe6dd18dc9b: https://hydra.nixos.org/build/131015808

**Comment by peterhoeg** [comment]
Working here.
